types/index.d.ts

```diff
@@ -1589,12 +1589,16 @@ declare namespace Knex {
     alter(): ColumnBuilder;
     queryContext(context: any): ColumnBuilder;
   }
+  
+  interface ForeignKeyBuilder {
+    withKeyName(keyName: string): ColumnBuilder;
+  }
 
-  interface ForeignConstraintBuilder {
+  interface ForeignConstraintBuilder extends ForeignKeyBuilder {
     references(columnName: string): ReferencingColumnBuilder;
   }
 
-  interface MultikeyForeignConstraintBuilder {
+  interface MultikeyForeignConstraintBuilder extends ForeignKeyBuilder {
```

Comment:
With this implementation, only when foreign key is made with ```foreign``` method will they be able to use withKeyName. 
Since both interfaces contain `references(columnName:string):ReferencingColumnBuilder`, shoudn't the ```withkeyName```  be put into ```interface ReferencingColumnBuilder``` or at least have this interface extends another interface that have all those method so that when foreign key is made with only ```references``` method, they can also those method?